torch/fx/experimental/symbolic_shapes.py

```diff
@@ -442,6 +442,11 @@ def rebind_unbacked(
         assert shape_env is not None
         for raw_u0, path in bindings.items():
             u1 = pytree.key_get(result, path)
+
+            # We only care about rebinding unbacked SymInts
+            if not isinstance(u1, torch.SymInt):
+                continue
```

Comment:
You care about rebinding ALL unbacked things, but many of your SymFloats are not unbacked. The better test is just to check if there are any unbacked symbols in u1.